src/main/java/athena/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,39 @@
+package athena.commands;
+
+import athena.tasks.TaskList;
+import athena.ui.Ui;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a deadline command given to Athena by the user. When executed, sets a new
+ * deadline according to the given task name and due date.
+ */
+public class DeadlineCommand extends Command {
+    private final String taskName;
+    private final LocalDateTime dueDateTime;
+
+    /**
+     * Constructs a new DeadlineCommand instance according to the given parameters.
+     *
+     * @param taskName The name of the deadline task.
+     * @param dueDateTime The due date and time of the deadline task.
+     */
+    public DeadlineCommand(String taskName, LocalDateTime dueDateTime) {
+        this.taskName = taskName;
+        this.dueDateTime = dueDateTime;
+    }
+
+    /**
+     * Creates a new deadline task in the given TaskList and outputs the results to the
+     * given Ui.
```

Comment:
It might align better with coding standard if you write one full sentence on one line 😄 (if I remember correctly from CheckStyle)

Reply:
Yes, good catch, will try to keep my summary sentences succinct!